### PR TITLE
[IE Myriad] Added Gelu layer

### PR DESCRIPTION
**Pull request by nikita-kud**
#28705  

- [x] 1244

### CODE REVIEW

File: inference-engine/src/vpu/myriad_plugin/myriad_plugin.cpp [hunk 1]
```diff
@@ -37,8 +63,10 @@ ExecutableNetworkInternal::Ptr Engine::LoadExeNetworkImpl(
     parsedConfigCopy.update(config);
 
     auto clonedNetwork = cloneNetwork(network);
+    std::cout << "LoadExeNetworkImpl";
     if (auto function = clonedNetwork->getFunction()) {
-        ngraph::op::GenericIE::DisableReshape noReshape(function);
+        Transformation(clonedNetwork);
```
Comment (gladilov-gleb) [inline inference-engine/src/vpu/myriad_plugin/myriad_plugin.cpp:68]: That's incorrect. There should not be conversion to CNN representation before our FrontEnd.

Required points:

1. Common optimization must happen before DTS
1. At the moment of DTS it must be nGraph, not CNN
1. Common optimization do not happen again (in full scope at least) after DTS
1. Conversion to CNN happens exactly at the same moment as before (in FrontEnd before precision conversion)

File: inference-engine/src/vpu/graph_transformer/src/frontend/frontend.cpp [hunk 2]
```diff
@@ -371,13 +373,17 @@ ModelPtr FrontEnd::runCommonPasses(ie::ICNNNetwork& network, const UnsupportedLa
         VPU_LOGGER_SECTION(env.log);
 
         auto convertNetwork = [&convertedNetwork, &originalOrConvertNetwork]() {
+            const auto transformations_callback = [](const std::shared_ptr<const ::ngraph::Node> &node) -> bool {
```
Comment (gladilov-gleb) [inline inference-engine/src/vpu/graph_transformer/src/frontend/frontend.cpp:376]: Name it predicate, please.
Also, comment describing why we don't want to process Gelu could be useful here

Comment (nikita-kud) [reply inference-engine/src/vpu/graph_transformer/src/frontend/frontend.cpp:376]: As you could see, I did it. But then I saw the name of the argument:
```
explicit ConvertOpSet3ToOpSet2(const PassParam::param_callback & callback = PassParam::getDefaultCallback())
            : FunctionPass(), PassParam(callback) {}
```  
Do you think this is more consistent than `callback`?  Also, it seems to me that this `callback `only asks the caller a question, and does not mean the order of calling certain functions.

> Also, comment describing why we don't want to process Gelu could be useful here  
It's a good idea. Something like? 
`We are notify NGraph that Myriad plugin supports Gelu and this layer should not be split into multiple layers. `

Comment (gladilov-gleb) [reply inference-engine/src/vpu/graph_transformer/src/frontend/frontend.cpp:376]: `callback` - it something that is triggered after event happened, `predicate` - criteria for triggering event. So I thin `predicate` fits here. Indeed, in function it's named other way, but I see it as code style bug and don't think we should inherit it.

I thought something about:

```cpp
// disable GeLU decomposition
```

Comment (nikita-kud) [reply inference-engine/src/vpu/graph_transformer/src/frontend/frontend.cpp:376]: Ok, I would only say that all plugins use `transformations_callback`:)

Comment (ilya-lavrenov) [reply inference-engine/src/vpu/graph_transformer/src/frontend/frontend.cpp]: @gladilov-gleb this `function` is called after callback inside transformation is called, so it's a callback based on your description.

Comment (gladilov-gleb) [reply inference-engine/src/vpu/graph_transformer/src/frontend/frontend.cpp]: If it's called after transformation which way it prevents from GeLU decomposition?
Anyway, it returns true or false which defines will we run transformation or not, right? If my understand is correct here, I still see that as `predicate`

File: inference-engine/src/vpu/graph_transformer/src/frontend/frontend.cpp [hunk 3]
```diff
@@ -372,13 +374,18 @@ ModelPtr FrontEnd::runCommonPasses(ie::ICNNNetwork& network, const UnsupportedLa
         VPU_LOGGER_SECTION(env.log);
 
         auto convertNetwork = [&convertedNetwork, &originalOrConvertNetwork]() {
+            // disable GeLU decomposition
+            const auto transformations_predicate = [](const std::shared_ptr<const ::ngraph::Node> &node) -> bool {
```
Comment (Maxim-Doronin) [inline inference-engine/src/vpu/graph_transformer/src/frontend/frontend.cpp:378]: Nitpick: please, use camelCase

File: inference-engine/src/vpu/graph_transformer/src/frontend/frontend.cpp [hunk 4]
```diff
@@ -372,13 +374,18 @@ ModelPtr FrontEnd::runCommonPasses(ie::ICNNNetwork& network, const UnsupportedLa
...
+                return std::dynamic_pointer_cast<const ::ngraph::opset2::Gelu>(node) != nullptr;
```
Comment (Maxim-Doronin) [inline inference-engine/src/vpu/graph_transformer/src/frontend/frontend.cpp:379]: Why are you using opset2 if opset3 is currently relevant?

File: inference-engine/src/vpu/graph_transformer/src/stages/gelu.cpp [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    }
+
+    void serializeParamsImpl(BlobSerializer& serializer) const override {
+    }
+};
+
+}  // namespace
+
+void FrontEnd::parseGelu(const Model& model, const ie::CNNLayerPtr& layer, const DataVector& inputs, const DataVector& outputs) const {
+    IE_ASSERT(inputs.size() == 1);
```
Comment (Maxim-Doronin) [inline inference-engine/src/vpu/graph_transformer/src/stages/gelu.cpp:37]: Please, use VPU_THROW_UNLESS

File: inference-engine/src/vpu/graph_transformer/src/stages/gelu.cpp [hunk 6]
```diff
@@ -0,0 +1,43 @@
+// Copyright (C) 2018-2020 Intel Corporation
```
Comment (Maxim-Doronin) [inline inference-engine/src/vpu/graph_transformer/src/stages/gelu.cpp:1]: 2020 only

File: inference-engine/src/vpu/graph_transformer/src/stages/gelu.cpp [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+
+#include <vpu/frontend/frontend.hpp>
+
+#include <cmath>
+
+#include <vector>
+#include <limits>
+#include <memory>
+#include <set>
+#include <string>
```
Comment (Maxim-Doronin) [inline inference-engine/src/vpu/graph_transformer/src/stages/gelu.cpp:13]: Do we need all of these libs?

File: inference-engine/src/vpu/graph_transformer/src/frontend/frontend.cpp (hunk 4 above)
Comment (nikita-kud) [reply inference-engine/src/vpu/graph_transformer/src/frontend/frontend.cpp:379]: copy-paste)

**Comment by nikita-kud** [comment]
- [ ] 1229

**Comment by nikita-kud** [comment]
@Maxim-Doronin could you merge it

**Pull request merged**